### PR TITLE
Restore PR #36 synchronous bus route check flow

### PR DESCRIPTION
**Pull request by hoonly01**
## Summary
- restore the synchronous bus route-check flow from the second repository pull request 36 head commit `b0adaaf7d3a08667b71e29f22984591d8b0c3471`
- remove callback/background-task branching from the Kakao bus route webhook and route image endpoint
- route synchronous webhook responses through `BusNoticeService.get_route_check_response` and update the webhook regression test accordingly
- deduplicate the TOPIS control info URL behind a module constant while keeping the restored response text intact

## Testing
- `uv run pytest tests/test_bus_notice.py -q`
- `uv run pytest -q`
- `uv run python -m py_compile app/routers/bus_notice.py app/services/bus_notice_service.py tests/test_bus_notice.py`
- `git diff --check`
- `basedpyright app/routers/bus_notice.py app/services/bus_notice_service.py tests/test_bus_notice.py`

## Source of truth
- second repository pull request 36 head commit: `b0adaaf7d3a08667b71e29f22984591d8b0c3471`
- original change title text: `feat: 카카오톡 버스 노선 조회 동기 응답 방식으로 변경 (콜백 제거)`

Closes #151

## Release Notes

* **Bug Fixes**
  * Improved error handling in route check requests to gracefully handle missing date parameters.
  * Enhanced error messages to provide clearer user guidance while concealing internal technical details.

* **Tests**
  * Added comprehensive test coverage for edge cases and error scenarios in webhook operations.

### CODE REVIEW

File: app/services/bus_notice_service.py [hunk 1]
```diff
@@ -247,6 +247,176 @@ async def generate_route_image(cls, route_number: str, date_str: str) -> Optiona
...
         return image_url
 
+    @classmethod
+    async def get_route_check_response(cls, route_number: str, params: Dict) -> Dict:
+        """노선 통제 확인 후 카카오톡 동기 응답 템플릿 반환"""
+        try:
+            target_date = params.get('date', '').strip()
+            if not target_date:
+                target_date = cls.korean_date_string()
+
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply app/services/bus_notice_service.py:264]: Decision: implemented

Rationale: Must Fix. `date: null` 같은 부분 입력은 예외 경로로 빠지지 않고 기본 날짜 처리로 흡수되어야 합니다. 기존 구현은 동기 웹훅이 복구 가능한 입력에서도 오류 응답으로 떨어질 수 있었습니다.

Commit hash: f5cd4d2b56ffd5c44867f888d71e78042a2ad985

Concise summary of the change: `date` 값을 문자열일 때만 `.strip()` 하도록 바꿔 `null`/비문자열 입력을 안전하게 처리했고, `date: null` 요청 회귀 테스트를 추가했습니다.

Validation performed: uv run pytest tests/test_bus_notice.py; uv run ruff check app/services/bus_notice_service.py tests/test_bus_notice.py; uv run python -m compileall app tests; uv run basedpyright app/services/bus_notice_service.py tests/test_bus_notice.py (0 errors, warnings only)

File: app/services/bus_notice_service.py [hunk 2]
```diff
@@ -247,6 +247,176 @@ async def generate_route_image(cls, route_number: str, date_str: str) -> Optiona
...
+                                )
+                            }
+                        }
+                    ]
+                }
+            }
+
+        except Exception as e:
+            logger.error(f"노선 조회 중 오류: {e}")
+            return {
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply app/services/bus_notice_service.py]: Decision: implemented

Rationale: Optional. 핵심 기능 버그는 아니지만, 현재 범위 안에서 매우 저위험이며 웹훅 회귀 조사에 필요한 스택 트레이스를 남겨 디버깅 품질을 높입니다.

Commit hash: f5cd4d2b56ffd5c44867f888d71e78042a2ad985

Concise summary of the change: 동기 route-check 예외 처리에서 `logger.error(...)`를 `logger.exception(...)`로 바꿔 서버 로그에 스택 트레이스가 남도록 했습니다.

Validation performed: uv run pytest tests/test_bus_notice.py; uv run ruff check app/services/bus_notice_service.py tests/test_bus_notice.py; uv run python -m compileall app tests; uv run basedpyright app/services/bus_notice_service.py tests/test_bus_notice.py (0 errors, warnings only)

File: app/services/bus_notice_service.py [hunk 3]
```diff
@@ -247,6 +247,176 @@ async def generate_route_image(cls, route_number: str, date_str: str) -> Optiona
...
+                "version": "2.0",
+                "template": {
+                    "outputs": [
+                        {
+                            "simpleText": {
+                                "text": f"❌ 노선 {route_number}번\n\n시스템 오류가 발생했습니다: {str(e)[:50]}"
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (hoonly01) [reply app/services/bus_notice_service.py]: Decision: implemented

Rationale: Must Fix. 사용자 응답에 `str(e)`를 노출하면 내부 구현/오류 세부사항이 외부로 새어 나갈 수 있어 보안과 운영 측면에서 막아야 합니다.

Commit hash: f5cd4d2b56ffd5c44867f888d71e78042a2ad985

Concise summary of the change: 사용자 노출 오류 문구를 공통 재시도 안내로 바꾸고, 내부 예외 세부사항이 응답에 포함되지 않음을 검증하는 회귀 테스트를 추가했습니다.

Validation performed: uv run pytest tests/test_bus_notice.py; uv run ruff check app/services/bus_notice_service.py tests/test_bus_notice.py; uv run python -m compileall app tests; uv run basedpyright app/services/bus_notice_service.py tests/test_bus_notice.py (0 errors, warnings only)
